src/Imath/ImathVec.h

```diff
@@ -73,14 +123,39 @@ template <class T> class IMATH_EXPORT_TEMPLATE_TYPE Vec2
     /// Copy constructor
     IMATH_HOSTDEVICE constexpr Vec2 (const Vec2& v) IMATH_NOEXCEPT;
 
+    /// Move constructor
+    IMATH_HOSTDEVICE constexpr Vec2 (Vec2&& v) IMATH_NOEXCEPT;
+
+    /// public api constructor
+    IMATH_HOSTDEVICE constexpr Vec2 (const Storage& v) IMATH_NOEXCEPT;
+
+    /// bare C array constructor
+    IMATH_HOSTDEVICE constexpr Vec2 (const T v[2]) IMATH_NOEXCEPT;
+
+    /// initializer_list constructor
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Vec2 (std::initializer_list<T> v)
+        IMATH_NOEXCEPT;
+
     /// Construct from Vec2 of another base type
-    template <class S>
-    IMATH_HOSTDEVICE constexpr Vec2 (const Vec2<S>& v) IMATH_NOEXCEPT;
+    template <typename S, typename SStorage>
+    IMATH_HOSTDEVICE constexpr Vec2 (const Vec2<S, SStorage>& v) IMATH_NOEXCEPT;
```

Comment:
Why not call this "Construct from a Vec2 of any base type and storage" and then it subsumes at least two other constructors?

Reply:
yeah, I wasn't 100% sure where I was going to end up, we can probably elide some of the other constructors with that one, but we should agree that this is the best path first... now that I've written it all, I'm not sure I like the idea of templating on the storage type, I think it'd be better to do something more like what you did for the "foreign" vector import, and instead template based on the vector type as a whole, so:

template <typename V, imath_enable_if(it's a vec2/vec3/whatevs)>
function(V v)

which then allows all forms. (although maybe you still need to keep the reference vs. rvalue reference as separate, depending on what you're doing)